v2/lints/cabf_br/lint_ca_key_usage_not_critical.go

```diff
@@ -33,6 +33,12 @@ func (l *caKeyUsageNotCrit) Initialize() error {
 }
 
 func (l *caKeyUsageNotCrit) CheckApplies(c *x509.Certificate) bool {
+
+	// this check doesn't apply, if the certificate is not a server auth certificate
+	if !util.IsServerAuthCert(c) {
+		return false
+	}
+
 	return c.IsCA && util.IsExtInCert(c, util.KeyUsageOID)
```

Comment:
We can keep this code concise as well.

```suggestion
	// Key Usage is only required to be critical for Root CA and Subordinate CA certificates, and does not
	// currently apply to OCSP responder certificates.
	return c.IsCA && util.IsExtInCert(c, util.KeyUsageOID) && util.IsServerAuthCert(c)
```

This is also trying to remove the double negation from the original comment ("does **not** ... if **not**"), and tries to explain the exceptional situation (i.e. what isn't covered by the code), to make it more obvious what the code is functionally trying to accomplish.

Reply:
I think we will in a later point in time need a discussion, what makes a certificate a TLS capable certificate. The function `IsServerAuthCert` was introduced with [commit](https://github.com/zmap/zlint/pull/171/commits/2caf99bdc64da040f4877cebff2f199a17f59253) and is not justified with references to the BRGs, Root Store Policies or RFCs.

In the Mozilla Root Store Policy 1.1 we find:

> 2.  Intermediate certificates which have at least one valid, unrevoked chain up
>     to such a CA certificate and which are technically capable of issuing 
>     working server or email certificates. Intermediate certificates that are not 
>     considered to be technically capable will contain either:
> 
>     * an Extended Key Usage (EKU) extension which does not contain any of
>       these KeyPurposeIds: anyExtendedKeyUsage, id-kp-serverAuth,
>       id-kp-emailProtection; and/or:
>     * name constraints which do not allow Subject Alternative Names (SANs) of
>       any of the following types: dNSName, iPAddress, SRVName, rfc822Name
> 
> 3.  End-entity certificates which have at least one valid, unrevoked chain up
>     to such a CA certificate through intermediate certificates which are all in
>     scope, such end-entity certificates having either:
> 
>     * an Extended Key Usage (EKU) extension which contains one or more of these
>       KeyPurposeIds: anyExtendedKeyUsage, id-kp-serverAuth,
>       id-kp-emailProtection; or:
>     * no EKU extension.

In the BRGs 7.1.2.3 we have:

> f. `extKeyUsage` (required)
> 
>    Either the value `id-kp-serverAuth` [RFC5280] or `id-kp-clientAuth` [RFC5280] or both values MUST be present. `id-kp-emailProtection` [RFC5280] MAY be present. Other values SHOULD NOT be present. The value `anyExtendedKeyUsage` MUST NOT be present.
> 

and in 7.1.2.2

> g. `extkeyUsage` (optional/required)
> 
>    For Cross Certificates that share a Subject Distinguished Name and Subject Public Key with a Root Certificate operated in accordance with these Requirements, this extension MAY be present. If present, this extension SHOULD NOT be marked critical. This extension MUST only contain usages for which the issuing CA has verified the Cross Certificate is authorized to assert. This extension MAY contain the `anyExtendedKeyUsage` [RFC5280] usage, if the Root Certificate(s) associated with this Cross Certificate are operated by the same organization as the issuing Root Certificate.
> 
>    For all other Subordinate CA Certificates, including Technically Constrained Subordinate CA Certificates:
> 
>    This extension MUST be present and SHOULD NOT be marked critical[^**].
> 
>    For Subordinate CA Certificates that will be used to issue TLS certificates, the value `id-kp-serverAuth` [RFC5280] MUST be present. The value `id-kp-clientAuth` [RFC5280] MAY be present. The values `id-kp-emailProtection` [RFC5280], `id-kp-codeSigning` [RFC5280], `id-kp-timeStamping` [RFC5280], and `anyExtendedKeyUsage` [RFC5280] MUST NOT be present. Other values SHOULD NOT be present.
> 
>    For Subordinate CA Certificates that are not used to issue TLS certificates, then the value `id-kp-serverAuth` [RFC5280] MUST NOT be present. Other values MAY be present, but SHOULD NOT combine multiple independent usages (e.g. including `id-kp-timeStamping` [RFC5280] with `id-kp-codeSigning` [RFC5280]).

We should double check, if the `IsServerAuthCert` is reflecting this.